scripts/apps/authoring/authoring/previewModal.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import {Modal} from 'core/ui/components/Modal/Modal';
+import {ModalHeader} from 'core/ui/components/Modal/ModalHeader';
+import {ModalBody} from 'core/ui/components/Modal/ModalBody';
+import {ModalFooter} from 'core/ui/components/Modal/ModalFooter';
+
+function getFormattedDocument(url) {
+    return new Promise((resolve) => {
+        var xhr = new XMLHttpRequest();
```

Comment:
Is there a reason why `XMLHttpRequest` is used instead of the newer Fetch APIs (available since Chrome v42, with a polyfill option for older browsers)?

Reply:
I don't remember the exact issue I had, but I tried using `fetch` and felt like it was a too low-level API for what would have been simpler with XHR.